### ISSUE TITLE
Unable to document default argument on friend functions

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10846]
**Describe the bug**
Unable to document default argument on friend functions

```
/**
 * A docs
 */
class A
{
public:
    friend void fun(int i);
};

/**
 * fun docs
 */
void fun(int i = 0);
```

Does not show the default argument `int i =0`.  The standard prohibits default argument in friend function declaration.

> §8.3.6 If a friend declaration specifies a default argument expression, that declaration shall be a definition and shall be the only declaration of the function or function template in the translation unit.

It looks like Doxygen uses the friend declaration of the function instead of the definition.

**Screenshots**
1.8.9.1 - trunk (85eb55d1b32bfeb474be6485c88348ef5d5d7fd8)
![image](https://github.com/doxygen/doxygen/assets/32584126/c165d305-e27d-420b-ae9c-db88d17ee411)

The default argument `int i = 0` is not documented.

**To Reproduce**
[testcase.zip](https://github.com/doxygen/doxygen/files/15178320/testcase.zip)

**Expected behavior**
The default argument `int i = 0` is documented.

**Comment by doxygen** [comment]
@scott-zhong Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by albert-github** [comment]
In the file html/class_a.html I still don't see the initialization:

![image](https://github.com/doxygen/doxygen/assets/5223533/35f3fd13-4f0e-4d22-ba84-d2799dcd7c66)

**Comment by doxygen** [comment]
@albert-github Thanks for checking. I noticed I was testing with a modified version of the example, which make the fix appear to work. I just committed a second attempt.

**Comment by albert-github** [comment]
Looks OK now, also when using:
```
friend void fun(int j);
```
or
```
friend void fun(int);
```
the results look OK to me, i.e.

![image](https://github.com/doxygen/doxygen/assets/5223533/a301642a-d9db-4ecc-b8a8-c14ec1c775e1)

and

![image](https://github.com/doxygen/doxygen/assets/5223533/44d37c27-046b-4a88-8c09-6070a574aa78)

**Comment by scott-zhong** [comment]
@doxygen verified the fix. Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
